src/Executor/Promise/Adapter/ReactPromiseAdapter.php

```diff
@@ -52,8 +52,13 @@ public function createRejected(Throwable $reason): Promise
         return new Promise($promise, $this);
     }
 
-    public function all(array $promisesOrValues): Promise
+    public function all(iterable $promisesOrValues): Promise
     {
+        assert(
+            is_array($promisesOrValues),
+            'ReactPromiseAdapter::all() requires arg of array type'
+        );
```

Comment:
Does not seem necessary

Reply:
Reactphp uses array currently https://github.com/reactphp/promise/blob/234f8fd1023c9158e2314fa9d7d0e6a83db42910/src/functions.php#L78

Reply:
Ok, luckily there is `iterable_to_array()`

Reply:
That fails when key is not `int|string`